plugin.py

```diff
@@ -1,99 +1,87 @@
-                subprocess.check_call(popenArgs, shell=True)
-            else:
-                subprocess.check_call(popenArgs)
-            onExit()
-        except subprocess.CalledProcessError as error:
-            logAndShowMessage('LSP-json: Error while installing the server.', error)
-        return
-    thread = threading.Thread(target=runInThread, args=(onExit, popenArgs))
-    thread.start()
-    # returns immediately after the thread starts
-    return thread
+    server.setup()
 
 
-def is_node_installed():
-    return shutil.which('node') is not None
+def plugin_unloaded():
+    server.cleanup()
 
 
-def logAndShowMessage(msg, additional_logs=None):
-    print(msg, '\n', additional_logs) if additional_logs else print(msg)
-    sublime.active_window().status_message(msg)
+def is_node_installed():
+    return shutil.which('node') is not None
 
 
 class LspJSONPlugin(LanguageHandler):
+    def __init__(self) -> None:
+        self._default_schemas = []
+
     @property
     def name(self) -> str:
-        return 'lsp-json'
+        return PACKAGE_NAME.lower()
 
     @property
     def config(self) -> ClientConfig:
-        settings = sublime.load_settings("LSP-json.sublime-settings")
-        client_configuration = settings.get('client')
+        # Calling setup() also here as this might run before `plugin_loaded`.
+        # Will be a no-op if already ran.
+        # See https://github.com/sublimelsp/LSP/issues/899
+        server.setup()
+
+        configuration = self.migrate_and_read_configuration()
+
         default_configuration = {
-            "command": [
-                'node',
-                server_path,
-                '--stdio'
-            ],
-            "languages": [
-                {
-                    "languageId": "json",
-                    "scopes": ["source.json"],
-                    "syntaxes": [
-                        "Packages/JavaScript/JSON.sublime-syntax",
-                        "Packages/JSON/JSON.sublime-syntax"
-                    ]
-                }
-            ]
+            'enabled': True,
+            'command': ['node', server.binary_path, '--stdio'],
         }
-        default_configuration.update(client_configuration)
-        default_configuration['settings']['json']['schemas'] = schemas
-        return read_client_config('lsp-json', default_configuration)
+
+        default_configuration.update(configuration)
+
+        if not self._default_schemas:
+            schemas = ['schemas_extra.json', 'schemas.json']
+            for schema in schemas:
+                path = 'Packages/{}/{}'.format(PACKAGE_NAME, schema)
+                self._default_schemas.extend(json.loads(ResourcePath(path).read_text()))
+
+        default_configuration['settings'].setdefault('json', {})['schemas'] = self._default_schemas
+
+        return read_client_config(self.name, default_configuration)
+
+    def migrate_and_read_configuration(self) -> None:
```

Comment:
The return type is not correct`-> None`
You probably meant a dict

Reply:
Fixed but since I don't have any automatic checking for those, I'm bound to get this wrong. Is there any way to configure `pyls` to do mypy linting? Or rather, how to do it, as I know that it's supposed to support mypy.
(But then there are bigger errors in this project when trying to run mypy on it)

Reply:
> Is there any way to configure `pyls` to do mypy linting?

https://github.com/tomv564/pyls-mypy seems work (with `live_mode` enabled though)

Reply:
@rchl did you install pyls with `pip3 install python-language-server[all]` ? If you installed it with just `pip3 install python-language-server` (without the `[all]`) then it's missing a few hooks/plugins, among which pyls-mypy.

Reply:
`pyls-mypy` is not include in `python-language-server[all]`

Reply:
I just can't get it to work.
Installed both with pip3 from brew (`/usr/local/bin/pip3`) and one I have in osx (`/usr/bin/pip3`) but I'm still not sure if language server is seeing it...

Reply:
OK, got it to work.
`pyls` was started by Python 2 as it was what `/usr/local/bin/python` pointed at. Linking Python 3 to `/usr/local/bin/python` fixed it (almost, also had to `pip3 install future` for `pyls_mypy` plugin).

Reply:
does `pyls_mypy` highlight relative imports as errors on your side?

Reply:
Sort of... probably not for the same reason as for you as non-relative imports also have similar error:

![Screenshot 2020-03-18 at 11 54 15](https://user-images.githubusercontent.com/153197/76953635-50d84c80-690f-11ea-8812-37b86d299000.png)